h5netcdf/core.py

```diff
@@ -258,6 +271,28 @@ def _ensure_dim_id(self):
             if "_Netcdf4Dimid" in dim.attrs:
                 self._h5ds.attrs["_Netcdf4Dimid"] = dim.attrs["_Netcdf4Dimid"]
 
+    def _add_fillvalue(self, fillvalue, dtype):
+        """Add _FillValue attribute"""
+        # trying to create correct type of fillvalue
+        if self.dtype is str:
+            value = fillvalue
+        else:
+            # todo: this always checks for dtype.metadata
+            string_info = self._root._h5py.check_string_dtype(self.dtype)
+            enum_info = self._root._h5py.check_enum_dtype(self.dtype)
+            if (
+                string_info
+                and string_info.length is not None
+                and string_info.length > 1
+            ) or enum_info:
+                value = fillvalue
+            else:
+                value = self.dtype.type(fillvalue)
+
+        # need to use create-function in order
+        # to provide correct committed/named type
+        self.attrs._h5attrs.create("_FillValue", value, dtype=dtype)
```

Comment:
Specifying the dtype here (for user types this would be the committed type) will link/share the type in the attribute.

Reply:
Note to self: Might need to special case `_FillValue` in `attrs.py` `__setitem__` to write it with dtype (committed type) from variable in any case (user might overwrite `_FillValue` attribute)